pkg/logger/logger_test.go

```diff
@@ -0,0 +1,170 @@
+package logger_test
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/go-logr/logr"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/gbytes"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+
+	. "github.com/vmware-tanzu/cartographer/pkg/logger"
+)
+
+var _ = Describe("Logger", func() {
+	var (
+		err error
+		logOpt zap.Opts
+		controllerOutput io.Writer
+		logger logr.Logger
+		controllerBuffer *gbytes.Buffer
+	)
+
+	BeforeEach(func() {
+		controllerBuffer = gbytes.NewBuffer()
+		controllerOutput = io.MultiWriter(controllerBuffer, GinkgoWriter)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	JustBeforeEach(func() {
+		logger = zap.New(logOpt, zap.WriteTo(controllerOutput))
+	})
+
+	When("log output level is info", func() {
+		BeforeEach(func() {
+			logOpt, err = SetLogLevel("info")
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		DescribeTable("log enabled is correct",
+			func(loglevel int, expected bool) {
+				Expect(logger.V(loglevel).Enabled()).To(Equal(expected))
+			},
+			Entry("info", 0, true),
+			Entry("debug", 1, false),
+			Entry("trace", 2, false),
+		)
+
+		It("outputs info logs", func() {
+			aNicePhrase := "It's a beautiful day in the neighborhood"
+			logger.V(0).Info(aNicePhrase)
```

Comment:
I don't think we should specify a verbosity on the logger itself. logger.Info should log for Info and Debug logs. logger.Debug should log for Debug.

Reply:
If this is not available with current interface, I think we should write our own

Reply:
I think this is a fine idea and one that I've considered as well. I think that's a different story. But yeah, we would just have our own wrapper of the `logr.logger` interface that has two loggers, the original and then one that had the verbosity increased to V1.

Reply:
The logr README has an example of using `.V(int).Info`: https://github.com/go-logr/logr#but-i-really-want-to-use-a-format-string